### ISSUE TITLE
CouchDB setup: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.

### ISSUE THREAD

**Issue by arminus** [vrtmrz/obsidian-livesync#17]
So I've tried to set up CouchDB with a Caddy proxy and a LetsEncrypt cert, below's the docker compose. Access to Fauxton on the given address works, however, testing the connection from the plugin throws this error (my.domain.tld is actually ${COUCHDB_SERVER} as configured for Docker):

Access to fetch at 'https://my.domain.tld//' from origin 'app://obsidian.md' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.

Any ideas?

```
version: "3.7"

services:

  caddy:
    image: lucaslorentz/caddy-docker-proxy:ci-alpine
    container_name: caddy
    ports:
      - 80:80
      - 443:443
    environment:
      - CADDY_INGRESS_NETWORKS=caddy
    networks:
      - caddy
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./data/caddy:/data

  couchserver:
    image: couchdb
    ports:
      - "5984:5984"
    environment:
      - COUCHDB_USER=${COUCHDB_USER}
      - COUCHDB_PASSWORD=${COUCHDB_PW}
    volumes:
        - ./data/couchdb:/opt/couchdb/data
    networks:
      - caddy
    labels:
      caddy: ${COUCHDB_SERVER}
      caddy.header.Access-Control-Allow-Origin: "app://obsidian.md"
      caddy.header.Access-Control-Allow-Methods: "\"GET, POST, PATCH, PUT, DELETE, OPTIONS\""
      caddy.header.Access-Control-Allow-Credentials: "true"
      caddy.header.Access-Control-Allow-Headers: "Origin, Content-Type"
      caddy.header.-Server: ""
      caddy.reverse_proxy: "{{upstreams 5984}}"

  # https://stackoverflow.com/questions/63623075/how-to-run-commands-from-docker-compose-yml-to-start-couchdb
  initializer:
    image: curlimages/curl
    depends_on:
      - couchserver
    command: ["sh","-c","sleep 15 && curl -u ${COUCHDB_USER}:${COUCHDB_PW} -X PUT couchserver:5984/_users &&  curl -u ${COUCHDB_USER}:${COUCHDB_PW} -X PUT couchserver:5984/test"]
    networks:
      - caddy
```

**Comment by vrtmrz** [comment]
Thank you for asking me! This is the awesome docker configuration. almost perfect. (I use traefik instead of caddy)
But we have to configure the CouchDB a little more.

The schema "app" is not acceptable, so we have to set wildcard to here.

And the CouchDB's configuration is also needed.  
Set below in local.ini on /opt/couchdb/etc/ of image

```
[chttpd]
require_valid_user = true

[chttpd_auth]
require_valid_user = true

[httpd]
WWW-Authenticate = Basic realm="couchdb"
enable_cors = true

[cors]
origins = *
credentials = true
headers = accept, authorization, content-type, origin, referer
methods = GET, PUT, POST, HEAD, DELETE
```
If you don't want to use ini for the configuration, you can set these values by HTTP API. ([docs](https://docs.couchdb.org/en/stable/config/intro.html))
like below
```
curl -X PUT http://localhost:5984/_node/couchdb@127.0.0.1/_config/chttpd/require_valid_user -d '"true"'
```
But it's a little troublesome.
 
I'm really sorry for delaying documentations...

FYI:
You don't necessarily have to keep the data of CouchDB if you allow some traffics when your image rebooted.
It's multi-master replication, it will be restored from some devices.

**Comment by arminus** [comment]
Great, using your local.ini let's me connect ok (with the Test button). However, after that, I get hundreds of messages like this:

```
Obsidian:24.11.2021, 10:34:25->{
  "status": 400,
  "name": "bad_request",
  "message": "Only reserved document ids may start with underscore.",
  "error": true
}
```
I suppose this is coming from the _resources folder in my vault.

In Fauxton, I see that a database "obsydian_livesync_version" had been created, but it's empty - so apart from _resources, nothing else had been synced yet either...

**Comment by vrtmrz** [comment]
Oh... You're right.
If you have items that are named as like  _xxx under the root of the vault. It will fail to save.
I'll fix this issue.

**Comment by arminus** [comment]
I think that _resources folder was the default folder for attachments which was created by Obsidian originally some time ago...? Or maybe by my original import in to Obsidian? Really don't remember, too long ago...

**Comment by vrtmrz** [comment]
@arminus 
_resource folder is not in my vault, but we have to handle it properly.
fixed this issue on 0.1.19, please try it.

**Comment by arminus** [comment]
Tested successfully, _resource folder is synced - at least with the initial sync (see https://github.com/vrtmrz/obsidian-livesync/issues/19 - but this is unrelated)
